### PR TITLE
Add provider-neo

### PR DESCRIPTION
**Pull request by chenquanyu**
Add provider-neo

### CODE REVIEW

File: provider-neo/src/main/java/network.arkane.provider.neo/sign/NeoTransactionSignable.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.math.BigInteger;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+public class NeoTransactionSignable implements Signable {
+
+    private List<RawTransactionAttribute> attributes;
+    private List<RawTransactionInput> inputs;
```
Comment (Qkyrie) [inline provider-neo/src/main/java/network.arkane.provider.neo/sign/NeoTransactionSignable.java]: Is there a guide on how to construct these attributes, inputs, outputs and scripts?

File: provider-neo/pom.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            <artifactId>provider-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>network.arkane</groupId>
+            <artifactId>provider-api</artifactId>
+            <scope>test</scope>
+            <type>test-jar</type>
+        </dependency>
+
+<!--        <dependency>-->
```
[the remark this answers is not in the document]
Comment (chenquanyu) [reply provider-neo/pom.xml]: Hi Qkyrie,
The provider-api is used in unit test

File: provider-neo/pom.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<!--            <groupId>io.neow3j</groupId>-->
+<!--            <artifactId>core</artifactId>-->
+<!--            <version>1.0.10</version>-->
+<!--        </dependency>-->
+
+        <dependency>
```
[the remark this answers is not in the document]
Comment (chenquanyu) [reply provider-neo/pom.xml]: Hi 
The 1.0.10 version has some issue for maven, soI build the package by myself.
I will change the dependence to 1.0.11

File: provider-neo/src/main/java/network.arkane.provider.neo/bridge/NeoTransactionGateway.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                            .build();
+                } else {
+                    log.debug("Got error from Neo chain: {}", send.getError().getMessage());
+                    throw arkaneException()
+                            .errorCode("transaction.submit.Neo-error")
+                            .message(send.getError().getMessage())
+                            .build();
+                }
+            } else {
+                log.debug("Updating last nonce");
```
[the remark this answers is not in the document]
Comment (chenquanyu) [reply provider-neo/src/main/java/network.arkane.provider.neo/bridge/NeoTransactionGateway.java]: Yes, I will remove the extra logs

File: provider-neo/src/main/java/network.arkane.provider.neo/sign/NeoTransactionSignable.java (hunk 1 above)
Comment (chenquanyu) [reply provider-neo/src/main/java/network.arkane.provider.neo/sign/NeoTransactionSignable.java]: For now, the contract is still not convenient to construct, neow3j is working for it now.

The example from neow3j is here:
RawTransaction rawTx = RawTransaction.createContractTransaction(
                null,
                null,
                Arrays.asList(
                        new RawTransactionInput("9feac4774eb0f01ab5d6817c713144b7c020b98f257c30b1105062d434e6f254", 0)
                ),
                Arrays.asList(
                        new RawTransactionOutput(0, NEOAsset.HASH_ID, "100.0", "AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y"),
                        new RawTransactionOutput(1, NEOAsset.HASH_ID, "900.0", multiSigAddress)
                )
);

File: pom.xml [hunk 5]
```diff
@@ -22,6 +22,7 @@
         <module>provider-vechain</module>
         <module>blockcypher</module>
         <module>integrationtests</module>
+        <module>provider-neo</module>
     </modules>
 
     <distributionManagement>
```
Comment (Qkyrie) [inline pom.xml]: Please also add
```
      <dependency>
                <groupId>network.arkane</groupId>
                <artifactId>provider-neo</artifactId>
                <version>${project.version}</version>
            </dependency>
```

to the dependencyManagement

**Comment by Qkyrie** [comment]
To fix the build, also add 
```
        <dependency>
            <groupId>network.arkane</groupId>
            <artifactId>provider-neo</artifactId>
        </dependency>
```

to the pom.xml of the integrationtests module.

File: provider-neo/src/main/java/network.arkane.provider.neo/secret/generation/NeoSecretGenerator.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import network.arkane.provider.secret.generation.SecretGenerator;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+
+import io.neow3j.crypto.ECKeyPair;
+import io.neow3j.crypto.Keys;
+
+public class NeoSecretGenerator implements SecretGenerator<NeoSecretKey> {
```
Comment (Qkyrie) [inline provider-neo/src/main/java/network.arkane.provider.neo/secret/generation/NeoSecretGenerator.java]: `@Component`

File: provider-neo/src/main/java/network.arkane.provider.neo/wallet/exporting/NeoKeystoreExporter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public NeoSecretKey reconstructKey(String secret, String password) {
+        return neoWalletDecryptor.generateKey(GeneratedNeoWallet.builder()
+                .walletFile(JSONUtil.fromJson(secret, WalletFile.class))
+                .build(), password);
+    }
+
+    @Override
+    public SecretType type() {
+        return SecretType.ETHEREUM;
```
Comment (Qkyrie) [inline provider-neo/src/main/java/network.arkane.provider.neo/wallet/exporting/NeoKeystoreExporter.java]: NEO

**Pull request merged**